gen/moon.hoon

```diff
@@ -15,11 +12,18 @@
         ~
         ~
     ==
-=/  ran  (clan p.bec)
+=*  our  p.bec
+=/  ran  (clan:title our)
 ?:  ?=({?($earl $pawn)} ran)
   %-  produce
   :-  %tang  :_  ~
   leaf+"can't create a moon from a {?:(?=($earl ran) "moon" "comet")}"
-=/  mon  (mix (lsh 5 1 (end 5 1 eny)) p.bec)
-=/  tic  .^(@ /a/(scot %p p.bec)/tick/(scot %da now)/(scot %p mon))
-(produce [%tang leaf+"ticket: {<`@p`tic>}" leaf+"moon: {<`@p`mon>}" ~])
+=/  mon=ship  (mix (lsh 5 1 (end 5 1 eny)) our)
+=/  cub=acru:ames  (pit:nu:crub:crypto 512 eny)
```

Comment:
"First" or "last" is a matter of perspective, of course. In any case, `eny` *is* 512 bits.

But in general, `+pit:nu:crub:crypto` takes a bitwidth and seed, converts the bitwidth to bytewidth and uses that for a length-adjusted SHA-512 (`+shal`) of the seed. `+shal` in turn truncates its input to the specified bytewidth. So that would be the "first" bytes semantically, from the little end (or "last" visually, with %ux for instance).

Reply:
Pretty sure that means "tl;dr no entropy leakage". Good!

Reply:
Actually that tl;dr is wrong, since "`eny` *is* 512 bits" means all of the data gets sent into `shal`. 32 of those bits are known, lowering the amount of security bits we have here.

Assuming that *is* correct, we should first use `eny` to get the `cub`, then `shal` it to get a new `eny` to use for `mon`. Seems sane?